### PR TITLE
BOM-2644: Django 3.2 Support

### PR DESCRIPTION
**Pull request by mzulqarnain1**
## Description

- Added Django 3.2 Support & Testing
- Applied Django Code Mods & Pyupgrade 
- Make Upgrade

## Ticket Link

Link to the associated ticket
https://openedx.atlassian.net/browse/BOM-2644

### CODE REVIEW

**Comment by mzulqarnain1** [comment]
Tests on Django 3.0+ will start running after https://github.com/edx/course-discovery/pull/3073 is merged.

**Review by awais786 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,15 +11,8 @@ jobs:
       fail-fast: false
       matrix:
         python-version: ['py38']
-        django-version: ['django22']
+        django-version: ['django22', 'django30', 'django31', 'django32']
         status: ['']
-        include:
```
Comment (awais786) [inline .github/workflows/ci.yml:16]: Revert this change

**Comment by awais786** [comment]
@waheedahmed it make sense to appear django30,31 and 32 like this. Otherwise they appeared as green and give wrong impression.

**Review by waheedahmed — APPROVED** [review]
